content/articles/exception-handeling-in-windows-forms/index.md

```diff
@@ -0,0 +1,130 @@
+- Any input besides numbers will cause an error since we only accept numbers
+- Denominator has the value of 0 because we can't divide by zero
+
+### Fixing the code
+In order to fix the code and resolve the previous issues, we will use exception handling. The Try scope will take care of both of the issues and notify any following Catch scopes, the first catch scope will resolve the invalid inputs from the two text boxes and the second catch will resolve the issue of having zero as a value of the Denominator. The Finally scope is optional as mentioned before, but we can use it here to reset the value of the two text boxes for the next operation.
+```c#
+  private void button1_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                int numerator = Convert.ToInt32(textBox1.Text);
+                int denominator = Convert.ToInt32(textBox2.Text);
+
+                int Answer = numerator / denominator;
+
+                textBox3.Text = Answer.ToString();
+            }
+            catch(FormatException)
+            {
+                MessageBox.Show("Invalid input enter two numbers please");
+            }
+            catch (DivideByZeroException divideByZeroException)
+            {
+                MessageBox.Show(divideByZeroException.Message,"Can't divide by zero");
+
+            }
+            finally
+            {
+                textBox1.Text = "";
+                textBox2.Text = "";
+            }
+        }
+```
+This is how the message box will appear when an issue occurs:
+
+![Erroroutput](/engineering-education/exception-handling-in-windows-forms/Erroroutput.png)
+
+### Useful tip
+Only use the tools of exception handling when you need them, for example, the following code requires the date of birth of a user.
+
+![datenormal](/engineering-education/exception-handling-in-windows-forms/datenormal.png)
+
+```c#
+ private void button1_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                int day = Convert.ToInt32(Daybox.Text);
+                int year = Convert.ToInt32(Yearbox.Text);
+                string month = Monthbox.Text;
+
+                
+            }
+            catch (FormatException)
+            {
+                MessageBox.Show("Invalid input, enter a valid form of date please");
+            }          
+            finally
+            {
+                Daybox.Text = "";
+                Monthbox.Text = "";
+                Yearbox.Text = "";
+            }
+        }
+```
+the previous piece of code will run the code correctly without any errors because we used exception handling, but instead, we could just use the *DateTimePicker* tool from the Toolbox.
+
+![datetool](/engineering-education/exception-handling-in-windows-forms/datetool.png)
+
+In this case, you don't need to use any of the exception handling tools because the user is not allowed to enter anything from the keyboard and therefore the input will always be valid.

```

Comment:
```suggestion
In this case, you don't need to use any of the exception handling tools because the user is not allowed to enter anything from the keyboard. Thus the input will always be valid.
```